org.lflang/src/org/lflang/generator/ts/TSReactionGenerator.kt

```diff
@@ -464,11 +516,20 @@ class TSReactionGenerator(
                 }
             }
         } else {
+            // get reactions
             generatedReactions = reactor.reactions
         }
+        // get mutations
+        generatedMutations = reactor.reactions.stream().filter{it -> it.isMutation()}.collect(Collectors.toList())
 
         ///////////////////// Reaction generation begins /////////////////////
         // TODO(hokeun): Consider separating this out as a new class.
+        if (generatedMutations.isNotEmpty()) {
+            // Write the mutation itself
+            for (mutation in generatedMutations) {
+                reactionCodes.add(generateSingleReaction(reactor, mutation))
```

Comment:
Does this method account for the fact that the given reaction is a mutation? If so, then why are we handling mutations and reactions separately at all?

Reply:
My question is: why do we gather mutations separately from reactions if for both we end up calling `generateSingleReaction`, anyway?

Reply:
Yes, you're right. I was mistaken that the `Reaction` and `Mutation`  were separated.
I deleted this part. https://github.com/lf-lang/lingua-franca/pull/1318/commits/61bb0286a4433f3d8c16efa3ed4ba017c7d307b0